src/main/kotlin/org/rust/ide/inspections/fixes/AttachFileToModuleFix.kt

```diff
@@ -54,19 +50,39 @@ class AttachFileToModuleFix(
     override fun invoke(project: Project, file: PsiFile, startElement: PsiElement, endElement: PsiElement) {
         val rsFile = startElement as? RsFile ?: return
         val availableModules = findAvailableModulesForFile(project, rsFile)
-        if (availableModules.isEmpty()) return
-
-        if (availableModules.size == 1) {
-            insertFileToModule(rsFile, availableModules[0])
-        } else if (availableModules.size > 1) {
-            selectModule(rsFile, availableModules)?.let { insertFileToModule(rsFile, it) }
+        when (availableModules.size) {
+            0 -> return
+            1 -> insertFileToModule(rsFile, availableModules[0])
+            else -> selectModule(rsFile, availableModules)?.let { insertFileToModule(rsFile, it) }
         }
     }
 
     override fun startInWriteAction(): Boolean {
         return false
     }
 
+    private fun insertFileToModule(file: RsFile, targetFile: RsFile) {
+        val project = file.project
+        val factory = RsPsiFactory(project)
+
+        // if the filename is mod.rs, attach it's parent directory
+        val name = if (file.isModuleFile) {
+            file.virtualFile.parent.name
+        } else {
+            file.virtualFile.nameWithoutExtension
+        }
+
+        val modItem = factory.tryCreateModDeclItem(name)
+        if (modItem == null) {
+            project.showBalloon("Could not create `mod ${name}`", NotificationType.ERROR)
+            return
+        }
+
+        project.runWriteCommandAction(text, "inspection.DetachedFile.Fix", targetFile) {
```

Comment:
Yeah, some style guidelines for groupId's and string ids would be awesome. I don't really care how it looks, as long as there is a clear guideline. 

By the way, I'm not sure I correctly understand the function of groupId. It just allows actions to be merged in the log? When would it make sense to give the same groupId to different actions? Should we set it at all, or maybe passing `null` everywhere would be the best solution?

Reply:
With regard to AttachFileToModuleFix, in general fixes aren't associated specifically to inspections or intentions. A fix may be applicable in many contexts, so it is unclear what suffix and what class name should be provided.

Perhaps we should use directly the name of the fix, i.e. `fix.Rust.AttachFileToModule`?

Reply:
> Should we set it at all, or maybe passing null everywhere would be the best solution?

Actually, it's a very good question
I've looked at usage and yes, it seems it's only for merging in undo stack.
And as the plugin doesn't have even two calls in the same "action", I think it would be much simpler just not to pass `groupId` at all. Also, it allows us not to come up with any rules about naming